test/test_Utils.jl

```diff
@@ -0,0 +1,119 @@
+@static if ENV["JULIA_JUSTRELAX_BACKEND"] === "AMDGPU"
+    using AMDGPU
+elseif ENV["JULIA_JUSTRELAX_BACKEND"] === "CUDA"
+    using CUDA
+end
+using Test
+using JustRelax, JustRelax.JustRelax2D, JustRelax.DataIO
+const backend_JR = CPUBackend
+
+using ParallelStencil, ParallelStencil.FiniteDifferences2D
+@init_parallel_stencil(Threads, Float64, 2) #or (CUDA, Float64, 2) or (AMDGPU, Float64, 2)
+
+using JustPIC, JustPIC._2D
+# Threads is the default backend,
+# to run on a CUDA GPU load CUDA.jl (i.e. "using CUDA") at the beginning of the script,
+# and to run on an AMD GPU load AMDGPU.jl (i.e. "using AMDGPU") at the beginning of the script.
+const backend = CPUBackend # Options: CPUBackend, CUDABackend, AMDGPUBackend
+# const backend = CUDABackend # Options: CPUBackend, CUDABackend, AMDGPUBackend
+
+# Load script dependencies
+using GeoParams
+
+@testset "Utils" begin
+    @testset "Macros" begin
+        # Set up mock data
+        # Physical domain ------------------------------------
+        ly           = 1.0       # domain length in y
+        lx           = 1.0       # domain length in x
+        nx, ny, nz   = 4, 4, 4   # number of cells
+        ni           = nx, ny     # number of cells
+        igg          = IGG(init_global_grid(nx, ny, 1; init_MPI= true)...)
+        li           = lx, ly     # domain length in x- and y-
+        di           = @. li / ni # grid step in x- and -y
+        origin       = 0.0, -ly   # origin coordinates (15km f sticky air layer)
+        grid         = Geometry(ni, li; origin = origin)
+        (; xci, xvi) = grid
+
+        # 2D case
+        dst = "test_Utils"
+        stokes  = StokesArrays(backend_JR, ni)
+        thermal = ThermalArrays(backend_JR, ni)
+        take(dst)
+        @test isdir(dst)
+        rm(dst, recursive=true)
+
+        nxcell, max_xcell, min_xcell = 20, 32, 12
+        particles = init_particles(
+            backend, nxcell, max_xcell, min_xcell, xvi...)
+        # temperature
+        pT, pPhases      = init_cell_arrays(particles, Val(2))
+        time = 1.0
+        dt = 0.1
+
+        stokes.viscosity.η .= @fill(1.0)
+        stokes.V.Vy        .= @fill(10)
+        thermal.T          .= @fill(100)
+
+        args = (P = stokes.P, T = thermal.T)
+        tuple_args = (args.P, args.T)
+        @test detect_args_size(tuple_args) == (7, 5)
```

Comment:
Instead of exporting `detect_args_size` (and the other newly exported functions), just import them as `import JustRelax2D : fun1, fun2, etc`